test/distributed/optim/test_named_optimizer.py

```diff
@@ -0,0 +1,189 @@
+# Owner(s): ["oncall: distributed"]
+
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import unittest
+
+import torch
+import torch.nn as nn
+
+from torch.distributed.optim import NamedOptimizer
+
+
+class TestDummyModel(torch.nn.Module):
+    def __init__(self):
+        super(TestDummyModel, self).__init__()
+        torch.manual_seed(0)
+        self.net1 = nn.Sequential(nn.Linear(8, 16), nn.ReLU())
+        self.net2 = nn.Sequential(nn.Linear(16, 32), nn.ReLU())
+        self.net3 = nn.Linear(32, 64)
+        self.net4 = nn.Sequential(nn.ReLU(), nn.Linear(64, 8))
+
+    def forward(self, x):
+        return self.net4(self.net3(self.net2(self.net1(x))))
+
+
+class NamedOptimizerTest(unittest.TestCase):
+    def _compare_state_dict_group(self, group, named_group, compare_equal=True):
```

Comment:
nit: As a personal recommendation, I prefer to avoid default arguments when possible for internal methods. The only tradeoff is verbosity, but that verbosity can be helpful when reading.

For example below in `test_load_state_dict`, if you pass `False` as the value for `compare_equal` vs. if you pass nothing, it is not immediately obvious what that means. A developer would have to take the additional step to (re-)read this method.

In contrast, imagine if you wrote:
```
self._compare_state_dict_groups(group1, group2, assert_equal=False)
...
self._compare_state_dict_groups(group1, group2, assert_equal=True)
```
(I felt that `assert_equal` was clearer than `compare_equal`, but that is just personal choice.)
In this case, it is more clear what the method call is doing at the callsite without requiring an additional read of the `_compare_state_dict_group` implementation. I can tell the first call is checking `group1` and `group2` are _not equal_ and that the second call is checking that they are _equal_.

Reply:
Oh, I see. That makes sense. Will do.